src/hardware/tandy_sound.cpp

```diff
@@ -315,10 +315,13 @@ void TandyDAC::DmaCallback([[maybe_unused]] const DmaChannel*, DMAEvent event)
 
 void TandyDAC::ChangeMode()
 {
-	// Avoid under or overruning the mixer with invalid frequencies
-	// Typically frequencies are in the 8 to 22 Khz range
-	constexpr auto dac_min_freq_hz = 4900;
-	constexpr auto dac_max_freq_hz = 49000;
+	// Typical sample rates are 1.7. 5.5, 11, and rarely 22 KHz. Although
+	// several games (one being OutRun) set instantaneous rates above
+	// 100,000 Hz, we throw these out as invalid by using the following
+	// reasonable and sane maximum. There are no known instances where we
+	// should instead try to play such audio.
+	//
+	constexpr auto dac_max_sample_rate_hz = 49000;
```

Comment:
Nice explanation and I guess that max rate wouldn't really cause harm in a practical sense, but is it really necessary? What would happen if we just ditched it?

Reply:
I remember putting it in to eliminate to high frequency artifacts and harmonics; but forgot the exact cases.

There's another problem in that the Speex Resampler fails at really high rates. For example, a divisor of 1 means we'll request a sample rate of 3+ GHz; even a divisor of 8 is still in 450 KHz, which is still way beyond what Speex can handle.

I'm not willing to do this level of regression testing to make this change - but if anyone else wants to in a seperate PR; them the issue can bisect to them :-)

Reply:
Thanks for the explanation, I was just curious 😄 It would be useful to mention we need to cap the max rate to avoid such riduculously high sample rates that will upset Speex.

Reply:
Added - and much clearer (snippet):

```c++
// Although
// several games (one being OutRun) set instantaneous rates above
// 100,000 Hz, we throw these out as they can cause garbage high
// frequency harmonics and also cause problems for the Speex Resampler.
// For example, a clock divider value of 8 (which is valid) produces a
// 450 KHz sampling rate, which is way beyond what Speex can handle.
```

Reply:
Yep, that covers it and now it makes perfect sense why we need that upper limit 🚀

Reply:
Thanks for bashing through this with me :-)